File: Data Structures & Algorithms/three-integer-sum/submission-2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        results = []
        nums.sort()
        for i in range(len(nums) - 2):
            if nums[i] > 0:
                break
            if i > 0 and nums[i] == nums[i - 1]:
                continue

            target = -nums[i]
            j, k = i + 1, len(nums) - 1
            while j < k:
                cur_sum = nums[j] + nums[k]
                if cur_sum < target:
                    j += 1
                elif cur_sum > target:
                    k -= 1
                else:
                    candidate = [nums[i], nums[j], nums[k]]
                    if len(results) == 0 or results[len(results) - 1] != candidate:
                        results.append(candidate)
                    j += 1
        return results

File: Data Structures & Algorithms/maximum-subarray/submission-2.go
func maxSubArray(nums []int) int {
    max_sum, cur_sum := nums[0], 0
    for _, num := range nums {
		if cur_sum < 0 {
			cur_sum = 0
		}        
        cur_sum += num
		max_sum = max(max_sum, cur_sum)
    }
    return max_sum
}
